### PR TITLE
"calls once per day to get feedback"

### PR DESCRIPTION
**Pull request by Kasturi786**
To run project :
1. celery -A taskscheduler.tasks.celery_app worker -c 5 --loglevel=info -P gevent
2. celery -A taskscheduler.celery_app beat --loglevel=info
3.  comment  other schedulers if not work except  following ones
    'insert-new-user-in-the-feedback-table.': {
            'task': 'get_insert_into_feedback_table',
            'schedule': timedelta(minutes=30)
        },
        'call-once-a-day-by-checking-data-in-the-feedback-table.': {
            'task': 'get-feedback_after_first_call_from_new_user',
            'schedule': timedelta(days=1)
        }
4. tested  in test environmet for that have to set env variables  present in setting.py file

### CODE REVIEW

File: flaskapp/views/ivrflow.py [hunk 1]
```diff
@@ -420,17 +418,36 @@ def get_profile():
...
+    auth_token = os.environ['TWILIO_AUTH_TOKEN']
+    client = Client(account_sid, auth_token)
+
+    start = datetime.datetime.now()
+    calls = [call.from_ for call in client.calls.list() if
+             ((start.date() - call.start_time.date())
+              + (datetime.timedelta(days=0, hours=start.hour, minutes=start.minute, seconds=start.second)
+                 - datetime.timedelta(days=0, hours=call.start_time.time().hour, minutes=call.start_time.time().minute,
+                                      seconds=call.start_time.time().second))) < datetime.timedelta(days=7)
+             and TWILIO_MAIN_PHONE_NUMBER.strip() != call.from_]
```
Comment (scidam) [inline flaskapp/views/ivrflow.py:451]: `TWILIO_MAIN_PHONE_NUMBER` is assumed to be safe and well formed, i.e. we can guarantee that it doesn't include extra spaces

File: flaskapp/views/ivrflow.py [hunk 2]
```diff
@@ -420,17 +418,36 @@ def get_profile():
...
+    account_sid = os.environ['TWILIO_ACCOUNT_SID']
+    auth_token = os.environ['TWILIO_AUTH_TOKEN']
+    client = Client(account_sid, auth_token)
+
+    start = datetime.datetime.now()
+    calls = [call.from_ for call in client.calls.list() if
+             ((start.date() - call.start_time.date())
+              + (datetime.timedelta(days=0, hours=start.hour, minutes=start.minute, seconds=start.second)
+                 - datetime.timedelta(days=0, hours=call.start_time.time().hour, minutes=call.start_time.time().minute,
+                                      seconds=call.start_time.time().second))) < datetime.timedelta(days=7)
```
Comment (scidam) [inline flaskapp/views/ivrflow.py:450]: 7 is a magic number here; do we need to put it to the settings.py file?

File: flaskapp/views/ivrflow.py [hunk 3]
```diff
@@ -420,17 +418,36 @@ def get_profile():
...
+def get_from_phone_number_from_twilio_call_logs():
+    account_sid = os.environ['TWILIO_ACCOUNT_SID']
+    auth_token = os.environ['TWILIO_AUTH_TOKEN']
+    client = Client(account_sid, auth_token)
+
+    start = datetime.datetime.now()
+    calls = [call.from_ for call in client.calls.list() if
+             ((start.date() - call.start_time.date())
+              + (datetime.timedelta(days=0, hours=start.hour, minutes=start.minute, seconds=start.second)
+                 - datetime.timedelta(days=0, hours=call.start_time.time().hour, minutes=call.start_time.time().minute,
```
Comment (scidam) [inline flaskapp/views/ivrflow.py:449]: Not exactly sure about this construction; You're trying to retrieve all phone numbers in past 7 days without taking into account sec, min, hours; only days are matter, aren't they? does the following (below) line of code solves the problem?:

```
 calls = [call.from_ for call in client.calls.list() if ((start.date() - call.start_time.date()).days < 7]
```

File: flaskapp/models/ivr_models.py [hunk 4]
```diff
@@ -35,43 +34,42 @@
...
 )
 
 
 class User(DatesMixin, BaseModel):
     """ General user model """
 
-    id        = AutoField()                             # noqa: E221
-    username  = TextField(null=True)                    # noqa: E221
-    gender    = CharField(                              # noqa: E221
+    id = AutoField()  # noqa: E221
```
[the remark this answers is not in the document]
Comment (Kasturi786) [reply flaskapp/models/ivr_models.py:52]: Could we remove  # noqa: E221 from all code?

File: flaskapp/views/ivrflow.py (hunk 3 above)
Comment (Kasturi786) [reply flaskapp/views/ivrflow.py:449]: We are taking into account days, hours, min and sec.
1. timedelta(days=1) = 1 day, 0:00:00
2. z= start.date() - call.start_time.date()
3. y = (datetime.timedelta(days=0, hours=start.hour, minutes=start.minute, seconds=start.second)
                 - datetime.timedelta(days=0, hours=call.start_time.time().hour, minutes=call.start_time.time().minute,
                                      seconds=call.start_time.time().second))
4  z+y

example :
z= 2021-12-02(start.date)  2021-11-03(call.start_time.date )
z= 29 days, 0:00:00
 y = datetime(0, 20, 24 ,5 )(Today hours,minutes and second ) - datetime( 0, 21, 57 ,7) (hours,minutes and second when call is made)
y= -1 day, 22:26:58

Required lag time = z+y 

Required lag time = 28 days, 22:26:58
comparing 
if (Required lag time <=timedelta(days=7) )

Comment (Kasturi786) [reply flaskapp/views/ivrflow.py:449]: calls = [call.from_ for call in client.calls.list() if ((start.date() - call.start_time.date()).days < 7] it also works

File: flaskapp/models/ivr_models.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Kasturi786) [reply flaskapp/models/ivr_models.py]: done!

File: flaskapp/models/ivr_models.py [hunk 5]
```diff
@@ -133,33 +131,42 @@ class Meta:
 class UserToken(DatesMixin, BaseModel):
```
[the remark this answers is not in the document]
Comment (Kasturi786) [reply flaskapp/models/ivr_models.py]: done!

File: flaskapp/views/ivrflow.py [hunk 6]
```diff
@@ -420,17 +418,36 @@ def get_profile():
...
-    return {"message": "user not found", "failed": 400}
\ No newline at end of file
+    return {"message": "user not found", "failed": 400}
+
+
+# getting set of from_ numbers from twilio call logs  that are less than 7 days old
+def get_from_phone_number_from_twilio_call_logs():
+    account_sid = os.environ['TWILIO_ACCOUNT_SID']
+    auth_token = os.environ['TWILIO_AUTH_TOKEN']
+    client = Client(account_sid, auth_token)
```
[the remark this answers is not in the document]
Comment (Kasturi786) [reply flaskapp/views/ivrflow.py]: done!

File: flaskapp/views/ivrflow.py (hunk 1 above)
Comment (Kasturi786) [reply flaskapp/views/ivrflow.py]: done!

File: flaskapp/views/ivrflow.py (hunk 2 above)
Comment (Kasturi786) [reply flaskapp/views/ivrflow.py]: done!

File: flaskapp/views/ivrflow.py (hunk 3 above)
Comment (Kasturi786) [reply flaskapp/views/ivrflow.py]: done!

File: flaskapp/models/ivr_models.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Kasturi786) [reply flaskapp/models/ivr_models.py]: done!

File: flaskapp/models/ivr_models.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Kasturi786) [reply flaskapp/models/ivr_models.py]: done!

**Comment by Lina-Lex** [comment]
Great job team!!! It's amazin!

**Pull request merged**